packages/core/modules-sdk/src/medusa-app.ts

```diff
@@ -36,6 +39,16 @@ import { cleanGraphQLSchema } from "./utils"
 
 const LinkModulePackage = MODULE_PACKAGE_NAMES[Modules.LINK]
 
+declare module "@medusajs/types" {
+  export interface ModuleImplementations {
```

Comment:
very cool! I'm guessing you could do something similar in the starter and it would work for any custom module? 

q: does this override or append to the ModuleImplementations type?

Reply:
This will merge the interfaces, this is the approach I ve also provided to the community in medusa v1 for extending interfaces for customization when for example they want the product type to include the newly added props accross the code base

Reply:
It merges. So what you add in the starter will added on top and the existing one's will be replaced (only if the types are compatible with each other)